### PR TITLE
Fix/assembly opts

### PR DESCRIPTION
**Pull request by igorbrasileiro**
#### What problem is this solving?
Fix the `assemblyOption` structure that is passed to addToCart query. 

#### How to test it?
[Workspace](https://brasileirovtex--acctglobal.myvtex.com/mcproductrefid136/p)

1. Go into workspace
2. Click to customize combo 1
3. Select `Lanche Genérico`
4. Repeat steps 2 and 3 to the other assembly option
5. Click to add product to cart
6. Look at minicart if `Lanche Generico` appears

#### Screenshots or example usage:
**After**:
image
assemblyOptions that's passed to addToCart query
```json
"items": [
        {
            "id": 10104,
            "index": 0,
            "quantity": 1,
            "seller": "1",
            "options": [
                {
                    "assemblyId": "Family Box - 1_Combo2",
                    "id": "10101",
                    "quantity": 1,
                    "seller": "1",
                    "options": [
                        {
                            "assemblyId": "Combo 1_Lanche",
                            "id": "10091",
                            "quantity": 0,
                            "seller": "1"
                        },
                        {
                            "assemblyId": "Combo 1_Lanche",
                            "id": "10093",
                            "quantity": 1,
                            "seller": "1"
                        }
                    ]
                },
             ....
      }
    ]
```
**Before**:
image
```json
"items": [
        {
            "id": 10104,
            "index": 0,
            "quantity": 1,
            "seller": "1",
            "options": [
                {
                    "assemblyId": "Family Box - 1_Combo2",
                    "id": "10101",
                    "quantity": 1,
                    "seller": "1"
                },
                ...,
                {
                    "assemblyId": "Combo 1_Lanche",
                    "id": "10091",
                    "quantity": 0,
                    "seller": "1"
                },
                ...
            ]
        }
    ],
```

#### Describe alternatives you've considered, if any.
Refactor the structure of Assembly Options Groups and Items at the Product Customizer

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- Tests do `transformAssemblyOptions` function.
+
```
Comment (klzns) [inline CHANGELOG.md:12]:
```suggestion
```

File: react/__tests__/modules.test.ts [hunk 2]
```diff
@@ -114,6 +115,28 @@ describe('assemblyOptions module', () => {
...
+      expect(comboOne.options?.[0].id).toBe('10091')
+      expect(comboOne.options?.[0].quantity).toBe(0)
+      expect(comboOne.options?.[1].id).toBe('10093')
+      expect(comboOne.options?.[1].quantity).toBe(1)
+      // check combo two
+      expect(comboTwo.options).toHaveLength(2)
+      expect(comboTwo.options?.[0].id).toBe('10091')
+      expect(comboTwo.options?.[0].quantity).toBe(0)
+      expect(comboTwo.options?.[1].id).toBe('10093')
+      expect(comboTwo.options?.[1].quantity).toBe(1)
```
Comment (klzns) [inline react/__tests__/modules.test.ts:139]: This test file is a bit weird.

I would recommend moving this to a separate test (another `it`)

**Review by klzns — CHANGES_REQUESTED** [review]

**Comment by klzns** [comment]
It needs rebase with master

File: react/__tests__/modules.test.ts (hunk 2 above)
Comment (igorbrasileiro) [reply react/__tests__/modules.test.ts]: Fixed

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- Tests to `transformAssemblyOptions` function.
+
```
Comment (klzns) [inline CHANGELOG.md:12]:
```suggestion
```

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by klzns — APPROVED** [review]

**Review by icazevedo — APPROVED** [review]
Very clean solution :noice:

**Review by victorhmp — APPROVED** [review]

**Pull request closed without merging**